backend/infrahub/menu/generator.py

```diff
@@ -3,31 +3,55 @@
 from typing import TYPE_CHECKING
 
 from infrahub.core import registry
-from infrahub.core.branch import Branch  # noqa: TCH001
 from infrahub.core.protocols import CoreMenuItem
 from infrahub.log import get_logger
+from infrahub.permissions.local_backend import LocalPermissionBackend
 
 from .constants import FULL_DEFAULT_MENU
 from .models import MenuDict, MenuItemDict
 
 if TYPE_CHECKING:
     from infrahub.auth import AccountSession
+    from infrahub.core.branch import Branch
     from infrahub.database import InfrahubDatabase
 
 log = get_logger()
 
 
 def get_full_name(obj: CoreMenuItem) -> str:
-    return f"{obj.namespace.value}:{obj.name.value}"
+    return f"{obj.namespace.value}{obj.name.value}"
 
 
-# pylint: disable=too-many-branches
-async def generate_menu(
+async def generate_restricted_menu(
     db: InfrahubDatabase, branch: Branch, menu_items: list[CoreMenuItem], account: AccountSession | None = None
 ) -> MenuDict:
-    # FIXME temp hack to avoid pylint to complain
-    account = account  # noqa: PLW0127
+    menu = await generate_menu(db=db, branch=branch, menu_items=menu_items)
 
+    if not account:
+        return menu
```

Comment:
This part should never come into play `get_menu()` will always send an object but it might be an unauthenticated object. Even if that happens I think it could make sense to run the code below so that we'd also hide the admin menu for anonymous users. The function signature here could change `account: AccountSession | None = None` -> `account: AccountSession`